rdrf/rdrf/forms/dynamic/form_changes.py

```diff
@@ -0,0 +1,36 @@
+class FormChangesExtractor:
+
+    CDE_SEPARATOR = "____"
+
+    def __init__(self, registry_form, previous_data, dynamic_data):
+        self.registry_form = registry_form
+        self.previous_data = previous_data
+        self. dynamic_data = dynamic_data
+        self.allowed_cde = []
+        self.form_changes = {}
+        self.previous_values = {}
+
+    def extract_cde(self, value):
+        return value.split(self.CDE_SEPARATOR)[-1]
+
+    def _extract_cdes_from_multi_section_values(self, multi_section_values):
+        result = set()
+        for vals in multi_section_values.values():
+            for d in vals:
+                cdes = [self.extract_cde(r) for r in d]
+                for x in cdes:
+                    result.add(x)
```

Comment:
`result.update(cdes)` instead of looping through and adding them one-by-one?

Reply:
Updated.